src/cdk/testing/testbed/fake-events/event-objects.ts

```diff
@@ -79,11 +79,11 @@ export function createPointerEvent(type: string, clientX = 0, clientY = 0,
  * Creates a browser TouchEvent with the specified pointer coordinates.
  * @docs-private
  */
-export function createTouchEvent(type: string, pageX = 0, pageY = 0) {
+export function createTouchEvent(type: string, pageX = 0, pageY = 0, clientX = 0, clientY = 0) {
   // In favor of creating events that work for most of the browsers, the event is created
   // as a basic UI Event. The necessary details for the event will be set manually.
   const event = document.createEvent('UIEvent');
-  const touchDetails = {pageX, pageY};
+  const touchDetails = {pageX, pageY, clientX, clientY};
```

Comment:
The mdc-slider checks the `clientX` and `clientY` of the `targetTouches[0]` when calculating the new value of the slider.

https://github.com/material-components/material-components-web/blob/master/packages/mdc-slider/foundation.ts#L372-L374